pkg/image/interfaces.go

```diff
@@ -26,6 +26,8 @@ type ICache interface {
 	// GetCompatibleArchitecturesSet takes an image reference. a list of secrets and the client to the cluster and
 	// returns a set of architectures that are compatible with the image reference.
 	GetCompatibleArchitecturesSet(ctx context.Context, imageReference string, skipCache bool, secrets [][]byte) (sets.Set[string], error)
+	// clearCache purges the cache of all entries
+	ClearCache()
```

Comment:
```suggestion
	// storeGlobalPullSecret takes a pull secret and stores it in the ImageFacade. It will be used by the controller
	// in charge of watching the global pull secret and to store it in the ImageFacade's relevant private field.
	// Then, the ImageFacade will be responsible for consuming it during the inspection.
storeGlobalPullSecret(pullSecret []byte)
```

and remove it at lines 34-38